doc/tools/docgram/fullGrammar

```diff
@@ -2471,14 +2524,22 @@ rewriter: [
 | natural [ "?" | LEFTQMARK ] constr_with_bindings_arg
 | natural constr_with_bindings_arg
 | constr_with_bindings_arg
+| "!" constr_with_bindings
+| [ "?" | LEFTQMARK ] constr_with_bindings
+| lnatural "!" constr_with_bindings
+| lnatural [ "?" | LEFTQMARK ] constr_with_bindings
+| lnatural constr_with_bindings
+| constr_with_bindings
 ]
 
 oriented_rewriter: [
 | orient_rw rewriter
+| ltac2_orient rewriter
```

Comment:
In some cases (again grammar rules for tactics), the rules are repeated but with other non-terminal names, making it harder to verify that these are exactly the same rules. That being said, I think this particular example is a case where you were the one to rename the non-terminal, to avoid duplicate names.